frontend/src/domains/admin/CreateRoomModal/hooks/useCategorySelection.ts

```diff
@@ -1,17 +1,45 @@
-import { useState, useCallback } from 'react';
-import { CategoryListType } from '@/constants/categoryList';
+import { useState, useCallback, useEffect } from 'react';
+import { CategoryListType, CATEGORY_LIST } from '@/constants/categoryList';
 import { MAX_CATEGORIES } from '@/constants/maxCategories';
 
 export interface SelectedCategoryItem {
   icon: string | React.ReactNode;
   category: CategoryListType;
 }
 
-export function useCategorySelection() {
+interface UseCategorySelectionProps {
+  initialCategories?: CategoryListType[];
+}
+
+export function useCategorySelection({
+  initialCategories,
+}: UseCategorySelectionProps = {}) {
   const [selectedCategories, setSelectedCategories] = useState<
     SelectedCategoryItem[]
   >([]);
 
+  useEffect(() => {
+    if (initialCategories && initialCategories.length > 0) {
+      const initialSelectedCategories: SelectedCategoryItem[] =
```

Comment:
```suggestion
  if (!initialCategories || initialCategories.length === 0) return;

  const initialSelectedCategories: SelectedCategoryItem[] =
  ...
```

요렇게 earlyReturn으로 중첩문을 제거할수도 있을 것 같아요
근데 귀찮으면 안해도 댐 그냥 생각나서 적어봄